integrationtest/creds_integration_test.go

```diff
@@ -463,3 +463,99 @@ func TestCreds_generated_role_rules(t *testing.T) {
+
+	type testCase struct {
+		roleConfig                  map[string]interface{}
+		credsConfig                 map[string]interface{}
+		expectedCredsCreateErrIsNil bool
+	}
+
+	tests := map[string]testCase{
+		"allowed_kubernetes_namespaces to * and kubernetes_namespace to test": {
+			roleConfig: map[string]interface{}{
+				"allowed_kubernetes_namespaces": []string{"*"},
+				"service_account_name":          "sample-app",
+			},
+			credsConfig: map[string]interface{}{
+				"kubernetes_namespace": "test",
+			},
+			expectedCredsCreateErrIsNil: true,
+		},
+		"allowed_kubernetes_namespaces to a single namespace, allowed_kubernetes_namespace_selector to empty," +
+			" and kubernetes_namespace omitted": {
+			roleConfig: map[string]interface{}{
+				"allowed_kubernetes_namespaces": []string{"test"},
+				"service_account_name":          "sample-app",
+			},
+			credsConfig:                 nil,
+			expectedCredsCreateErrIsNil: true,
+		},
+		"allowed_kubernetes_namespaces to * and kubernetes_namespace omitted": {
+			roleConfig: map[string]interface{}{
+				"allowed_kubernetes_namespaces": []string{"*"},
+				"service_account_name":          "sample-app",
+			},
+			credsConfig:                 nil,
+			expectedCredsCreateErrIsNil: false,
+		},
+		"allowed_kubernetes_namespaces to a single namespace, allowed_kubernetes_namespace_selector to nonempty," +
+			" and kubernetes_namespace omitted": {
+			roleConfig: map[string]interface{}{
+				"allowed_kubernetes_namespaces":         []string{"test"},
+				"allowed_kubernetes_namespace_selector": `{"matchExpressions": [{"key": "target", "operator": "In", "values": ["integration-test"]}, {"key": "nonexistantlabel", "operator": "DoesNotExist", "values": []}]}`,
+				"service_account_name":                  "sample-app",
+			},
+			credsConfig:                 nil,
+			expectedCredsCreateErrIsNil: false,
+		},
+		"allowed_kubernetes_namespaces to empty, allowed_kubernetes_namespace_selector to nonempty," +
+			"kubernetes_namespace omitted": {
+			roleConfig: map[string]interface{}{
+				"allowed_kubernetes_namespace_selector": `{"matchExpressions": [{"key": "target", "operator": "In", "values": ["integration-test"]}, {"key": "nonexistantlabel", "operator": "DoesNotExist", "values": []}]}`,
+				"service_account_name":                  "sample-app",
+			},
+			credsConfig:                 nil,
+			expectedCredsCreateErrIsNil: false,
+		},
+		"allowed_kubernetes_namespaces to more than one specified, kubernetes_namespace omitted": {
+			roleConfig: map[string]interface{}{
+				"allowed_kubernetes_namespaces": []string{"test", "foo"},
+				"service_account_name":          "sample-app",
+			},
+			credsConfig:                 nil,
+			expectedCredsCreateErrIsNil: false,
+		},
+	}
+	i := 0
+	for n, tc := range tests {
+		t.Run(n, func(t *testing.T) {
+			roleName := fmt.Sprintf("testrole-%d", i)
+			_, err = client.Logical().Write(path+"/roles/"+roleName, tc.roleConfig)
+			assert.NoError(t, err)
```

Comment:
nit: The test can't meaningfully continue if this check fails, so `require` is a slightly better fit

```suggestion
			require.NoError(t, err)
```